burn-book/src/basic-workflow/model.md

```diff
@@ -51,14 +60,96 @@ There are two major things going on in this code sample.
    functionalities. Like other attributes often used in Rust, such as `Clone`, `PartialEq` or
    `Debug`, each field within the struct must also implement the `Module` trait.
 
-2. Note that the struct is generic over the `Backend` trait. The backend trait abstracts the
-   underlying low level implementations of tensor operations, allowing your new model to run on any
-   backend. Contrary to other frameworks, the backend abstraction isn't determined by a compilation
-   flag or a device type. This is important because you can extend the functionalities of a specific
-   backend (see [backend extension section](../advanced/backend-extension)), and it allows for an
-   innovative [autodiff system](../building-blocks/autodiff.md). You can also change backend during
-   runtime, for instance to compute training metrics on a cpu backend while using a gpu one only to
-   train the model. In our example, the backend in use will be determined later on.
+   <details>
+   <summary><strong>🦀 Trait</strong></summary>
+
+   Traits are a powerful and flexible Rust language feature. They provide a way to define shared
+   behavior for a particular type, which can be shared with other types.
+
+   A type's behavior consists of the methods called on that type. Since all `Module`s should
+   implement the same functionality, it is defined as a trait. Implementing a trait on a particular
+   type usually requires the user to implement the defined behaviors of the trait for their types,
+   though that is not the case here as explained above with the `derive` attribute. Check out the
+   [explainer below](#derive-attribute) to learn why.
+
+   For more details on traits, take a look at the
+   [associated chapter](https://doc.rust-lang.org/book/ch10-02-traits.html) in the Rust Book.
+   </details><br>
+
+   <details id="derive-attribute">
+   <summary><strong>🦀 Derive Attribute</strong></summary>
+
+   The `derive` attribute allows traits to be implemented easily by generating code that will
```

Comment:
Not only for traits, you can add arbitrary code based on the provided type, but you can't modify the existing code; you can only append new elements. Using the attribute `#[my_macro]`, you can effectively update the provided code. Derive is frequently employed to recursively implement traits, where the implementation consists of the composition of all fields.